File: lossdiafaq/services/calculator/flame.py
import math

from lossdiafaq import static


class FlameCalculator:
    """A calculator for Flame stats
    
    Attributes
    ----------
    level : :class:`int`
        The level of the gear being tested

    item_eflame_min_stats : :class:`int`
        The minimum range of the eternal flame stats

    item_eflame_max_stats : :class:`int`
        The maximum range of the eternal flame stats

    item_pflame_min_stats : :class:`int`
        The minimum range of the powerful flame stats

    item_pflame_max_stats : :class:`int`
        The maximum range of the powerful flame stats

    overall_eflame_min_stats : :class:`int`
        The minimum range of the eternal flame stats for overalls

    overall_eflame_max_stats : :class:`int`
        The maximum range of the eternal flame stats for overalls

    overall_pflame_min_stats : :class:`int`
        The minimum range of the powerful flame stats for overalls

    overall_pflame_max_stats : :class:`int`
        The maximum range of the powerful flame stats for overalls
    """
    def __init__(self, level: int) -> None:
        self.level = level

        self.item_eflame_min_stats = 0
        self.item_eflame_max_stats = 0
        self.item_pflame_min_stats = 0
        self.item_pflame_max_stats = 0

        self.overall_eflame_min_stats = 0
        self.overall_eflame_max_stats = 0
        self.overall_pflame_min_stats = 0
        self.overall_pflame_max_stats = 0

    def calculate(self):
        """Calculates the stat ranges for each flame and item type
        
        Overalls get double flame stats
        """
        self.item_eflame_min_stats = (math.ceil(self.level / 20) + 1) * static.LOSSDIA_FLAME_EFLAME_MIN_RANGE
        self.item_eflame_max_stats = (math.ceil(self.level / 20) + 1) * static.LOSSDIA_FLAME_EFLAME_MAX_RANGE
        self.item_pflame_min_stats = (math.ceil(self.level / 20) + 1) * static.LOSSDIA_FLAME_PFLAME_MIN_RANGE
        self.item_pflame_max_stats = (math.ceil(self.level / 20) + 1) * static.LOSSDIA_FLAME_PFLAME_MAX_RANGE

        self.overall_eflame_min_stats = self.item_eflame_min_stats * 2
        self.overall_eflame_max_stats = self.item_eflame_max_stats * 2
        self.overall_pflame_min_stats = self.item_pflame_min_stats * 2
        self.overall_pflame_max_stats = self.item_pflame_max_stats * 2

        return

File: server/internal/server/listeners/flame.go
package listeners

import (
	"fmt"

	"github.com/jaczerob/server/internal/calculator"
	"github.com/jaczerob/server/internal/server"
	"github.com/jaczerob/server/internal/utils"
)

type Flame struct{}

var _ server.CommandListener = (*Flame)(nil)

func (f *Flame) Name() string {
	return "flame"
}

func (f *Flame) Run(c *server.Command) (r *server.ReturnData, err error) {
	level, err := c.Args.GetFloat(0)
	if err != nil {
		return
	}

	flame := calculator.NewFlameCalculator(level)
	flame.Calculate()

	r = &server.ReturnData{
		Embeds: []server.Embed{
			{
				Title:       "Flames Calculator",
				Description: fmt.Sprintf("The flame stats for a level %s item. Overall stats are double that of normal items.", utils.FormatFloat(level)),
				Fields: []server.EmbedField{
					{
						Name:   "Eternal Flame Stat Range",
						Value:  fmt.Sprintf("Overall: %s - %s\nNot Overall: %s - %s", utils.FormatFloat(flame.OverallEFlameMinStats), utils.FormatFloat(flame.OverallEFlameMaxStats), utils.FormatFloat(flame.ItemEFlameMinStats), utils.FormatFloat(flame.ItemEFlameMaxStats)),
						Inline: true,
					},
					{
						Name:   "Powerful Flame Stat Range",
						Value:  fmt.Sprintf("Overall: %s - %s\nNot Overall: %s - %s", utils.FormatFloat(flame.OverallPFlameMinStats), utils.FormatFloat(flame.OverallPFlameMaxStats), utils.FormatFloat(flame.ItemPFlameMinStats), utils.FormatFloat(flame.ItemPFlameMaxStats)),
						Inline: true,
					},
				},
			},
		},
	}
	//		"ItemEFlameMinStats":    flameCalculator.ItemEFlameMinStats,
	//		"ItemEFlameMaxStats":    flameCalculator.ItemEFlameMaxStats,
	//		"ItemPFlameMinStats":    flameCalculator.ItemPFlameMinStats,
	//		"ItemPFlameMaxStats":    flameCalculator.ItemPFlameMaxStats,
	//		"OverallEFlameMinStats": flameCalculator.OverallEFlameMinStats,
	//		"OverallEFlameMaxStats": flameCalculator.OverallEFlameMaxStats,
	//		"OverallPFlameMinStats": flameCalculator.OverallPFlameMinStats,
	//		"OverallPFlameMaxStats": flameCalculator.OverallPFlameMaxStats,
	//	}, nil
	return
}
